__tests__/article.test.js

```diff
@@ -0,0 +1,111 @@
+const app = require('.././app');
+const request = require('supertest');
+const db = require('../db/connection');
+const { setDefaultHighWaterMark } = require('supertest/lib/test');
+const seed = require('../db/seeds/seed');
+const seedDATA = require('../db/data/test-data/index')
+const fs = require('fs');
+const path = require('path');
+
+afterAll(() => db.end());
+beforeEach(()=>{
+  return seed(seedDATA);
+})
+
+describe('/api/articles/:article_id', () => {
+    test('GET 200: responds with an article object with the correct properties', () => {
+      return request(app)
+        .get('/api/articles/1')
+        .expect(200)
+        .then(({ body: { article } }) => {
+          expect(article).toEqual(
+            expect.objectContaining({
+              author: expect.any(String),
+              title: expect.any(String),
+              article_id: expect.any(Number),
+              body: expect.any(String),
+              topic: expect.any(String),
+              created_at: expect.any(String),
+              votes: expect.any(Number),
+              article_img_url: expect.any(String),
+            })
+          );
+        });
+    });
+  
+    test('GET 404: responds with an error when the article_id does not exist', () => {
+      return request(app)
+        .get('/api/articles/9999')
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.msg).toBe('Article not found');
+        });
+    });
+  
+    test('GET 400: responds with an error when the article_id is invalid', () => {
+      return request(app)
+        .get('/api/articles/not-a-number')
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe('Invalid article ID');
+        });
+    });
+  });
+
+  describe('/api/articles/:article_id/comments', () => {
+    test('GET 200: responds with an array of comments for the given article_id', () => {
+      return request(app)
+        .get('/api/articles/1/comments')
+        .expect(200)
+        .then(({ body: { comments } }) => {
+          expect(comments).toBeInstanceOf(Array);
+          expect(comments).toHaveLength(11);
+          comments.forEach((comment) => {
+            expect(comment).toEqual(
+              expect.objectContaining({
+                comment_id: expect.any(Number),
+                votes: expect.any(Number),
+                created_at: expect.any(String),
+                author: expect.any(String),
+                body: expect.any(String),
+                article_id: 1,
+              })
+            );
+          });
+          for (let i = 0; i < comments.length - 1; i++) {
+            expect(new Date(comments[i].created_at).getTime()).toBeGreaterThanOrEqual(
+              new Date(comments[i + 1].created_at).getTime()
+            );
+          }
```

Comment:
This is not how we expect you to check the order, please refer to the jest-sorted docs for a nicer alternative for this